doozer/doozerlib/metadata.py

```diff
@@ -401,11 +402,121 @@ async def get_latest_build(self, **kwargs):
             return await self.get_latest_brew_build_async(**kwargs)
 
         elif self.runtime.build_system == 'konflux':
-            raise NotImplementedError
+            return await self.get_latest_konflux_build(**kwargs)
 
         else:
             raise ValueError(f'Invalid value for --build-system: {self.runtime.build_system}')
 
+    async def get_pinned_konflux_build(self, el_target: int):
+        """
+        Return the assembly pinned NVR.
+        Under 'is' for RPMs, we expect 'el7' and/or 'el8', etc. For images, just 'nvr'.
+        """
+
+        is_config = self.config['is']
+
+        if self.meta_type == 'rpm':
+            if el_target is None:
+                raise ValueError(
+                    f'Expected el_target to be set when querying a pinned RPM component {self.distgit_key}')
+            is_nvr = is_config[f'el{el_target}']
+            if not is_nvr:
+                self.logger.warning('No pinned NVR found for RPM %s and target %s', self.rpm_name, el_target)
+                return None
+
+        else:
+            # The image metadata (or, more likely, the current assembly) has the image
+            # pinned. Return only the pinned NVR. When a child image is being rebased,
+            # it uses get_latest_build to find the parent NVR to use (if it is not
+            # included in the "-i" doozer argument). We need it to find the pinned NVR
+            # to place in its Dockerfile.
+            # Pinning also informs gen-payload when attempting to assemble a release.
+            is_nvr = is_config.nvr
+            if not is_nvr:
+                raise ValueError(f'Did not find nvr field in pinned Image component {self.distgit_key}')
+
+        # strict means raise an exception if not found.
+        return await self.runtime.konflux_db.get_build_record_by_nvr(is_nvr, strict=True)
+
+    async def get_latest_konflux_build(self, assembly: Optional[str] = None,
+                                       outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS,
```

Comment:
Any use case to return a build with outcome other than SUCESS?

Reply:
As usual, I followed the prototype we already had in `get_latest_brew_build`. Looks like we never pass in something different from a successful build in there too, so I would either change both or None to keep them as similar as possible

Reply:
If so, maybe always searching for SUCCESS build is better choice to keep the logic simpler and less confusing.

Reply:
Done with f19fd0546a3df6a2ddacdcffb5bbfc8697de5ef3.

Reply:
@thegreyd can you please double check if this looks good to you? Although I don't see cases where we search for anything different from COMPLETED it might have slipped

Reply:
actually, scan-sources searches for failed builds [here](https://github.com/openshift-eng/art-tools/blob/main/doozer/doozerlib/metadata.py#L745), so it looks like that param must stay. and for the new `get_latest_konflux_build`, it will be needed once we decide to swap out `konflux_db.get_latest_build`